tests/functional/common/init.sh

```diff
@@ -1,5 +1,30 @@
 # shellcheck shell=bash
 
+# for shellcheck
+: "${test_nix_conf_dir?}" "${test_nix_conf?}"
+
+if isTestOnNixOS; then
+
+  mkdir -p "$test_nix_conf_dir" "$TEST_HOME"
+
+  export NIX_USER_CONF_FILES="$test_nix_conf_dir/nix.conf"
+  mkdir -p "$test_nix_conf_dir" "$TEST_HOME"
+  ! test -e "$test_nix_conf"
```

Comment:
```suggestion
  (! test -e "$test_nix_conf")
```

Reply:
IIUC we need this when the negated command is involved in a pipe, which is not the case here.
I suppose it's a good habit to do it regardless?

Not sure what to put in the commit message otherwise.